python/pdo/submitter/ccf/ccf_submitter.py

```diff
@@ -0,0 +1,589 @@
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] # this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def ccl_update(self,
+        channel_keys,
+        contract_enclave_id,
+        enclave_signature,
+        contract_id,
+        message_hash,
+        current_state_hash,
+        previous_state_hash,
+        dependency_list,
+        **extra_params):
+
+        tx_method = "ccl_update"
+        verb = "update"
+        contract_code_hash =[]
+
+        dependencies = []
+        for dependency in dependency_list :
+            temp = dict()
+            temp['contract_id'] = dependency['contract_id']
+            temp['state_hash'] = crypto.base64_to_byte_array(dependency['state_hash'])
+            temp['state_hash_for_sign'] = dependency['state_hash']
+            dependencies.append(temp)
+
+        tx_params = PayloadBuilder.build_update_contract_state_transaction_from_data(
+            verb,
+            contract_enclave_id,
+            crypto.base64_to_byte_array(enclave_signature),
+            contract_id,
+            current_state_hash,
+            previous_state_hash,
+            dependencies,
+            message_hash,
+            contract_code_hash,
+            nonce = channel_keys
+            )
+
+        try:
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] #this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def get_enclave_info(self,
+        enclave_id):
+
+        tx_method = "verify_enclave_registration"
+        tx_params = PayloadBuilder.build_verify_enclave_from_data(enclave_id)
+
+        enclave_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+        # verify ccf signature
+        message = enclave_info["verifying_key"]
+        message+= enclave_info["encryption_key"]
+        message+= enclave_info["proof_data"]
+        message+= enclave_info["last_registration_block_context"]
+        message+= enclave_info["owner_id"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, enclave_info["signature"]):
+            raise Exception("Invalid signature on Get Enclave Info from CCF Ledger")
+
+        return enclave_info
+
+# -----------------------------------------------------------------
+    def get_contract_info(self,
+        contract_id):
+
+        tx_method = "verify_contract_registration"
```

Comment:
Echoing Bruno's comment above. The method name here is misleading, and should probably match the function name you've given it here "get_contract_info".

Reply:
get contract info is the external API, the tx_method is the ccf rpc handler, and the cleint does not see this at all. I dont see why the two should be same. If you look at the ccf implementation of the rpc, it is essentially checking if the contract was registered or not. so the name makes sense from for pdo tp.

Reply:
Whether the contract is registered is just a sanity check of the RPC handler, isn't it? The main purpose of the RPC handler is still to retrieve the contract info and return it.

Reply:
I can also argue the other way that retrieving the info is essential to check if it was registered or not. getting data is part of any check-operation. So get data is very generic & conveys little information.

Reply:
Naming the RPC method `get_contract_info` would convey the appropriate information, and does not diminish the need for checking whether the data exists in the first place. In any case, this suggestion was never meant to turn into a discussion about function semantics, really just suggesting a small name change.